src/client/Microsoft.Identity.Client/TelemetryCore/Http/HttpTelemetryManager.cs

```diff
@@ -78,12 +75,12 @@ public string GetLastRequestHeader()
 
                 if (ev.ContainsKey(MsalTelemetryBlobEventNames.RegionDiscovered))
                 {
-                    string region = ev[MsalTelemetryBlobEventNames.RegionDiscovered];
-
                     if (!firstFailure)
                         platformFields.Append(",");
 
-                    platformFields.Append(region);
+                    platformFields.Append(ev.RegionDiscovered);
+                    platformFields.Append(",");
+                    platformFields.Append(ev.RegionSource);
```

Comment:
>platformFields.Append(ev.RegionSource); [](start = 20, length = 39)

seems like you are missing a check on whether regionsource is available?

Reply:
If the region is discovered, the region source will be there and there is a check to see if the region is discovered.

---
In reply to: [526623194](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2199#discussion_r526623194) [](ancestors = 526623194)